### PR TITLE
wip first task

### PR DESCRIPTION
**Pull request by NikolayGGorbunov**
without tests

### CODE REVIEW

File: mapbox_task/first.rb [hunk 1]
```diff
@@ -1,13 +1,52 @@
...
-3) Если груз больше 1 метра кубического и его вес превышает 10кг, то цена перевозки ровна 3 рубля за километр.
-Тесты c использованием гемов RSpec и VCR
-Для запросов использовать класс Net::HTTP
-*/
+require 'json'
+require 'pry'
+require 'net/http'
+
+class DeliveryBoy
+  def self.get_cities_cords(city1, city2)
```
Comment (Sh1mensoka) [inline mapbox_task/first.rb:6]: Why don't we make this method private?

File: mapbox_task/first.rb [hunk 2]
```diff
@@ -1,13 +1,52 @@
...
+    response = Net::HTTP.get(uri)
+    par = JSON.parse(response)
+    raise "City not found" if par["features"].size < 2
+    if par["features"][0]["matching_place_name"].downcase.include?(city1)
+      city1_cord = par["features"][0]["center"]
+      city2_cord = par["features"][1]["center"]
+    else
+      city1_cord = par["features"][1]["center"]
+      city2_cord = par["features"][0]["center"]
+    end
```
Comment (Sh1mensoka) [inline mapbox_task/first.rb:19]: Does the order of coords really matter for distance calculation?

File: mapbox_task/first.rb [hunk 3]
```diff
@@ -1,13 +1,52 @@
...
+    [city1_cord,city2_cord]
+  end
+
+  def self.get_distance(cords1, cords2)
```
Comment (Sh1mensoka) [inline mapbox_task/first.rb:23]: Same as below, should be private method

File: mapbox_task/first.rb [hunk 4]
```diff
@@ -1,13 +1,52 @@
...
+    uri = URI("https://api.mapbox.com/directions/v5/mapbox/driving/#{cords1[0]}%2C#{cords1[1]}%3B#{cords2[0]}%2C#{cords2[1]}?alternatives=false&geometries=geojson&overview=simplified&steps=false&access_token=pk.eyJ1Ijoibmlrb2xhaWdvcmJ1bm92IiwiYSI6ImNsNm5veDg1ZDAyd2Yzam1sbnprZnAwYTYifQ.Ix2UzqHEHVU0E1-jqXsB4Q")
+    response = Net::HTTP.get(uri)
+    par = JSON.parse(response)
+    raise "No route" if par["code"].include?("NoSegment") || par["code"].include?("NoRoute")
+    distance = par["routes"].first["distance"]
+    (distance/1000).round
+  end
+
+  def self.package_type(wgh,lng,wth,hth)
```
Comment (Sh1mensoka) [inline mapbox_task/first.rb:32]: Same as below, should be private method

File: mapbox_task/first.rb [hunk 5]
```diff
@@ -1,13 +1,52 @@
...
+    if wgh < 0 || lng <= 0 || wth <= 0 || hth <= 0
+      raise "Wrong arguments"
+    end
+    cubic = lng*wth*hth
+    if cubic < 1000000
+      return 1
+    elsif wgh < 10
```
Comment (Sh1mensoka) [inline mapbox_task/first.rb:39]: What will happen if package's weight will be exactly 10?

File: spec/first_spec.rb [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    context "given size more then 1 cubic meter and weight less then 10 kg(float)" do
+      let(:weight) {3.21}
+      let(:length) {200}
+      let(:width) {200}
+      let(:height) {200}
+      it "return 2" do
+        expect(instance).to eq(2)
+      end
+    end
+  end
```
Comment (Sh1mensoka) [inline spec/first_spec.rb:66]: This case can be tested in previous tests by passing, for example, 10.0

File: spec/first_spec.rb [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+  describe "#get_cities_cords" do
+    subject(:instance) {described_class.get_cities_cords(city1, city2)}
```
Comment (Sh1mensoka) [inline spec/first_spec.rb:69]: nit: subject's name is not required when you use only one

File: spec/first_spec.rb [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+    context "given two different valid cities in reverse" do
+      let(:city1) {"moscow"}
+      let(:city2) {"krasnodar"}
+      it "return city2 to city1 cords right" do
+        VCR.use_cassette "get cities raw moscow-krasnodar" do
+          expect(instance[0]).to eq([37.61778, 55.75583])
+          expect(instance[1]).to eq([38.98333, 45.03333])
+        end
+      end
+    end
```
Comment (Sh1mensoka) [inline spec/first_spec.rb:97]: Is this case relevant if we only use coords to calculate distance between them?

File: spec/first_spec.rb [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+        end
+      end
+      it "return right distance in km" do
+        VCR.use_cassette "get distance krasnodar-moscow" do
+          expect(instance).to eq(1349)
+        end
+      end
+    end
+
+    context "given cords, but there's no way to get them" do
```
Comment (Sh1mensoka) [inline spec/first_spec.rb:137]: nit: you should name contexts more transparent to others, for example "when given incorrect coords"

File: spec/first_spec.rb [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+    end
+
+    context "given incorrect city names" do
+      let(:data) {{weight: 1, length:1, width:1, height:1, city1_name:"bjgdhbsjhgbsdg", city2_name:"sojdamsdbasfdvasf"}}
+      it "raise 'city not found' error" do
+        VCR.use_cassette "get cities raw no city" do
+          expect{instance}.to raise_error(RuntimeError, "City not found")
+        end
+      end
+    end
```
Comment (Sh1mensoka) [inline spec/first_spec.rb:188]: Duplicates test from https://github.com/NikolayGGorbunov/Trainee_repo/pull/1/files#diff-03c80ad53efeb8cacc643e95e9560386c7a130775b1ccbaa00151a89e323936dR107

File: spec/first_spec.rb [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+
+    context "given two same city names" do
+      let(:data) {{weight: 1, length:1, width:1, height:1, city1_name:"moscow", city2_name:"moscow"}}
+      it "raise 'same city' error" do
+        expect{instance}.to raise_error(RuntimeError, "Choose different cities")
+      end
+    end
```
Comment (Sh1mensoka) [inline spec/first_spec.rb:195]: Also a duplicate

Comment (NikolayGGorbunov) [reply spec/first_spec.rb]: Fixed all previous remarks , expect subjects name (but I'll call them better next time)

File: mapbox_task/first.rb [hunk 12]
```diff
@@ -1,13 +1,49 @@
...
-3) Если груз больше 1 метра кубического и его вес превышает 10кг, то цена перевозки ровна 3 рубля за километр.
-Тесты c использованием гемов RSpec и VCR
-Для запросов использовать класс Net::HTTP
-*/
+require 'json'
+require 'pry'
+require 'net/http'
+
+class DeliveryBoy
+  private
```
Comment (Sh1mensoka) [inline mapbox_task/first.rb:6]: Best practice is place public methods first (no need to use "public" directive) and after that place private methods.
Example:
class SomeClass
    def ...
    end
    ...
    private
    def ...
    end
 end

Comment (NikolayGGorbunov) [reply mapbox_task/first.rb:13]: Thanks for note. Done

**Review by Sh1mensoka — APPROVED** [review]

File: mapbox_task/first.rb [hunk 13]
```diff
@@ -1,13 +1,48 @@
...
+    (distance/1000).round
+  end
+
+  def self.package_type(weight,length,width,height)
+    if weight < 0 || length <= 0 || width <= 0 || height <= 0
+      raise "Wrong arguments"
+    end
+    cubic = length*width*height
+    if cubic < 1000000
+      return 1
```
Comment (NadezhdaGlin) [inline mapbox_task/first.rb:41]: why returned 1? what is this?

Comment (NikolayGGorbunov) [reply mapbox_task/first.rb:41]: It's a package_type. I used as a mul-parameter when calculate price (rows 9-10)

File: spec/first_spec.rb [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+    context "given fake cities" do
+      let(:city1) {"bjgdhbsjhgbsdg"}
+      let(:city2) {"sojdamsdbasfdvasf"}
+      it "raise 'city not found' error" do
+        VCR.use_cassette "get cities raw no city" do
+          expect{instance}.to raise_error(RuntimeError, "City not found")
+        end
+      end
+    end
+  end
```
Comment (NadezhdaGlin) [inline spec/first_spec.rb:95]: i think, you can also check if one city is entered

**Review by NadezhdaGlin — APPROVED** [review]
Good Luck

**Pull request merged**